common/persistence/sql/execution_tasks.go

```diff
@@ -146,7 +146,7 @@ func (m *sqlExecutionStore) getHistoryImmediateTasks(
 
 	rows, err := m.Db.RangeSelectFromHistoryImmediateTasks(ctx, sqlplugin.HistoryImmediateTasksRangeFilter{
 		ShardID:            request.ShardID,
-		CategoryID:         categoryID,
+		CategoryID:         int32(categoryID),
```

Comment:
nit: not sure if int can be used in sqlplugin interface as well.

Reply:
That should be fine. We can convert to/from int32, int64 wherever category ids are serialized or deserialized (e.g. protos / DB objects). However, outside of that, there's no reason for it to be anything other than an int.